### PR TITLE
Add CodeQL Analysis

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
This PR adds GitHub codeql analysis

### CODE REVIEW

**Review by daniel-heppner-ibigroup — APPROVED** [review]

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# For most projects, this workflow file will not need changing; you simply need
```
Comment (binh-dam-ibigroup) [inline .github/workflows/codeql-analysis.yml:1]: Add a comment saying the file was initially generated by github.

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "dev", master ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "dev" ]
+  schedule:
+    - cron: '23 11 * * 4'
```
Comment (binh-dam-ibigroup) [inline .github/workflows/codeql-analysis.yml:21]: Is the intent to run checks periodically in addition to PR and PR updates to `dev`?

**Review by binh-dam-ibigroup — APPROVED** [review]
LGTM, only wondering about whether periodicity of the checks is absolutely needed.

File: .github/workflows/codeql-analysis.yml (hunk 2 above)
Comment (miles-grant-ibigroup) [reply .github/workflows/codeql-analysis.yml:21]: It seems like it! Which makes sense -- even if we don't make changes new vulnerabilities might be discovered

Comment (binh-dam-ibigroup) [reply .github/workflows/codeql-analysis.yml:23]: Good point!

**Pull request merged**